### ISSUE TITLE
[JENKINS-10234] Junit result archiver getting stuck for a long time in concurrent builds

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/junit-plugin#900]
<p>When reaching the end of a build with jUnit results, possibly when the job is allowed to run concurrently, we are frequently seeing our system get stuck on "Recording test results".</p>

<p>Looking at the thread list, I see the following:<br/>
"Executor #9 for master : executing Run_Manual_SOAK #242 : waiting for Check point JUnit result archiving on Run_Manual_SOAK #241<br/>
java.lang.Object.wait(Native Method)<br/>
java.lang.Object.wait(Object.java:502)<br/>
hudson.model.Run$Runner$CheckpointSet.waitForCheckPoint(Run.java:1266)<br/>
hudson.model.Run.waitForCheckpoint(Run.java:1234)<br/>
hudson.model.CheckPoint.block(CheckPoint.java:144)<br/>
hudson.tasks.junit.JUnitResultArchiver.perform(JUnitResultArchiver.java:159)<br/>
hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:19)<br/>
hudson.model.AbstractBuild$AbstractRunner.perform(AbstractBuild.java:663)<br/>
hudson.model.AbstractBuild$AbstractRunner.performAllBuildSteps(AbstractBuild.java:638)<br/>
hudson.model.AbstractBuild$AbstractRunner.performAllBuildSteps(AbstractBuild.java:616)<br/>
hudson.model.Build$RunnerImpl.post2(Build.java:161)<br/>
hudson.model.AbstractBuild$AbstractRunner.post(AbstractBuild.java:585)<br/>
hudson.model.Run.run(Run.java:1399)<br/>
hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)<br/>
hudson.model.ResourceController.execute(ResourceController.java:88)<br/>
hudson.model.Executor.run(Executor.java:145)<br/>
Executor #9 for master : executing Run_Manual_SOAK #242 : waiting for Check point JUnit result archiving on Run_Manual_SOAK #241"</p>

<p>All the stuck jobs are in the same place. They do eventually come unstuck, but can spend a long time (hours and sometimes a day or so) in this state.<br/>
Machine load average is at 0.23 0.22 0.21. </p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dannystaple.jpg" title="dannystaple's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dannystaple">dannystaple</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-10234" target="_blank">Junit result archiver getting stuck for a long time in concurrent builds</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Blocker
<li><b>component(s)</b>: junit-plugin
<li><b>label(s)</b>: checkpoint, concurrent
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2014-07-21T17:47:58+00:00
<li><b>votes</b>: 15
<li><b>watchers</b>: 30
<li><b>imported</b>: 2025-11-26
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>When reaching the end of a build with jUnit results, possibly when the job is allowed to run concurrently, we are frequently seeing our system get stuck on "Recording test results".</p>

<p>Looking at the thread list, I see the following:
"Executor #9 for master : executing Run_Manual_SOAK #242 : waiting for Check point JUnit result archiving on Run_Manual_SOAK #241
java.lang.Object.wait(Native Method)
java.lang.Object.wait(Object.java:502)
hudson.model.Run$Runner$CheckpointSet.waitForCheckPoint(Run.java:1266)
hudson.model.Run.waitForCheckpoint(Run.java:1234)
hudson.model.CheckPoint.block(CheckPoint.java:144)
hudson.tasks.junit.JUnitResultArchiver.perform(JUnitResultArchiver.java:159)
hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:19)
hudson.model.AbstractBuild$AbstractRunner.perform(AbstractBuild.java:663)
hudson.model.AbstractBuild$AbstractRunner.performAllBuildSteps(AbstractBuild.java:638)
hudson.model.AbstractBuild$AbstractRunner.performAllBuildSteps(AbstractBuild.java:616)
hudson.model.Build$RunnerImpl.post2(Build.java:161)
hudson.model.AbstractBuild$AbstractRunner.post(AbstractBuild.java:585)
hudson.model.Run.run(Run.java:1399)
hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)
hudson.model.ResourceController.execute(ResourceController.java:88)
hudson.model.Executor.run(Executor.java:145)
Executor #9 for master : executing Run_Manual_SOAK #242 : waiting for Check point JUnit result archiving on Run_Manual_SOAK #241"</p>

<p>All the stuck jobs are in the same place. They do eventually come unstuck, but can spend a long time (hours and sometimes a day or so) in this state.
Machine load average is at 0.23 0.22 0.21. </p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Jenkins 1.415, Hudson 1.393. Both on Fedora, Tomcat 6. x86_64.</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>jayee:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-10234?focusedId=152861&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-152861">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We're also having this problem of different running times, as we sometimes are skipping a job in a chain of jobs.</p>

<p>Danny, have you found any workaround for this?</p>

<p>Temporarily disable the checkPoint waiting on certain jobs would help us. If the JUnitResultArchiver were a plugin it would simplify forking the feature.</p>
</pre>
</details>

</ul>
</details>

<p>We're also having this problem of different running times, as we sometimes are skipping a job in a chain of jobs.</p>

<p>Danny, have you found any workaround for this?</p>

<p>Temporarily disable the checkPoint waiting on certain jobs would help us. If the JUnitResultArchiver were a plugin it would simplify forking the feature.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>test4ever:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-10234?focusedId=183446&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-183446">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I was redirected from another issue to this one. My problem is the result recording takes forever for some of my builds. The job in question is a parametrized concurrent job.</p>
</pre>
</details>

</ul>
</details>

<p>I was redirected from another issue to this one. My problem is the result recording takes forever for some of my builds. The job in question is a parametrized concurrent job.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mcklaus.png" title="mcklaus's avatar" /> mcklaus:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-10234?focusedId=197228&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-197228">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>@clint axeda:
can you make a CPU sample using visualVM to check whether you experience high cpu-time consumption in CipherInputStream.fill_buffer() (sort by column "Self Time (CPU)")?</p>

<p>maybe we share the same root-cause: <a href="https://issues.jenkins.io/browse/JENKINS-22297" title="ssh-slave connection seems to have performance issues" class="issue-link" data-issue-key="JENKINS-22297"><del>JENKINS-22297</del></a></p>
</pre>
</details>

</ul>
</details>

<p>@​clint axeda:<br/>
can you make a CPU sample using visualVM to check whether you experience high cpu-time consumption in CipherInputStream.fill_buffer() (sort by column "Self Time (CPU)")?</p>

<p>maybe we share the same root-cause: <a href="https://issues.jenkins.io/browse/JENKINS-22297" title="ssh-slave connection seems to have performance issues" class="issue-link" data-issue-key="JENKINS-22297"><del>JENKINS-22297</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-10234?focusedId=206195&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206195">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Jesse Glick
Path:
 changelog.html
 core/src/main/java/hudson/tasks/junit/JUnitResultArchiver.java
<a href="http://jenkins-ci.org/commit/jenkins/90ff9f806fcac1a58f4bd40bfcc4ed5273ff116a" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/90ff9f806fcac1a58f4bd40bfcc4ed5273ff116a</a>
Log:
  <span class="error">&#91;FIXED JENKINS-10234&#93;</span> Removed checkpoint from JUnitResultArchiver.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 changelog.html<br/>
 core/src/main/java/hudson/tasks/junit/JUnitResultArchiver.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/90ff9f806fcac1a58f4bd40bfcc4ed5273ff116a" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/90ff9f806fcac1a58f4bd40bfcc4ed5273ff116a</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-10234&#93;</span> Removed checkpoint from JUnitResultArchiver.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogfood.png" title="dogfood's avatar" /> dogfood:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-10234?focusedId=206200&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206200">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/3535/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #3535</a>
     <span class="error">&#91;FIXED JENKINS-10234&#93;</span> Removed checkpoint from JUnitResultArchiver. (Revision 90ff9f806fcac1a58f4bd40bfcc4ed5273ff116a)</p>

<p>     Result = SUCCESS
Jesse Glick : <a href="https://github.com/jenkinsci/jenkins/commit/90ff9f806fcac1a58f4bd40bfcc4ed5273ff116a" class="external-link" target="_blank" rel="nofollow noopener">90ff9f806fcac1a58f4bd40bfcc4ed5273ff116a</a>
Files : </p>
<ul>
	<li>changelog.html</li>
	<li>core/src/main/java/hudson/tasks/junit/JUnitResultArchiver.java</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/3535/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #3535</a><br/>
     <span class="error">&#91;FIXED JENKINS-10234&#93;</span> Removed checkpoint from JUnitResultArchiver. (Revision 90ff9f806fcac1a58f4bd40bfcc4ed5273ff116a)</p>

<p>     Result = SUCCESS<br/>
Jesse Glick : <a href="https://github.com/jenkinsci/jenkins/commit/90ff9f806fcac1a58f4bd40bfcc4ed5273ff116a" class="external-link" target="_blank" rel="nofollow noopener">90ff9f806fcac1a58f4bd40bfcc4ed5273ff116a</a><br/>
Files : </p>
<ul>
	<li>changelog.html</li>
	<li>core/src/main/java/hudson/tasks/junit/JUnitResultArchiver.java</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-10234?focusedId=211825&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-211825">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Any change that we could get this backported to the 1.565.x LTS branch? It's one of the most annoying problems for our Jenkins production systems.</p>
</pre>
</details>

</ul>
</details>

<p>Any change that we could get this backported to the 1.565.x LTS branch? It's one of the most annoying problems for our Jenkins production systems.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-10234?focusedId=212210&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-212210">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>There is version 1.565.3 LTS to be scheduled for Oct 1st. Not sure when fixes are taking into consideration for LTS releases. Anything else beside the keyword we could try to get it in? The next major version bump for LTS will happen end of Oct, where this might be fixed.</p>
</pre>
</details>

</ul>
</details>

<p>There is version 1.565.3 LTS to be scheduled for Oct 1st. Not sure when fixes are taking into consideration for LTS releases. Anything else beside the keyword we could try to get it in? The next major version bump for LTS will happen end of Oct, where this might be fixed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-10234?focusedId=212591&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-212591">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Next LTS will be based on 1.580, and the 1.565.x line is done.</p>
</pre>
</details>

</ul>
</details>

<p>Next LTS will be based on 1.580, and the 1.565.x line is done.</p>
